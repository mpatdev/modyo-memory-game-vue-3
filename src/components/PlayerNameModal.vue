<script setup>
import GameModal from '@/components/core/GameModal.vue'
import GameTextBox from '@/components/core/GameTextBox.vue'
import GameButton from '@/components/core/GameButton.vue'

import { ref, defineEmits } from 'vue'

const emit = defineEmits([['update:modelValue']])
const playerName = ref('')

function startGame() {
  console.log('Send Start Game')
  emit('update:modelValue', playerName.value)
}
</script>

<template>
  <GameModal>
    <div class="flex w-full h-full justify-center items-center">
      <div class="w-full">
        <div class="mb-4 text-secondary text-lg">
          Hi, Welcome to Game Memory. Please type your name
        </div>
        <div>
          <GameTextBox
            v-model="playerName"
            placeholder="Enter your name"
          />
        </div>
        <div class="flex justify-end mt-4">
          <GameButton
            :disabled="!playerName"
            @click="startGame"
          >
            Start
          </GameButton>
        </div>
      </div>
    </div>
  </GameModal>
</template>