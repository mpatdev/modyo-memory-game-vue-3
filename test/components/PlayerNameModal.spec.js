import PlayerNameModal from '@/components/PlayerNameModal.vue'

import snapshotTest from '@test/snapshotTest'

snapshotTest(PlayerNameModal, 'PlayerNameModal')