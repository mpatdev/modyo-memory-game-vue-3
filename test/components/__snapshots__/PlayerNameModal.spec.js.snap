// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayerNameModal > Check Snapshot 1`] = `
"<div class=\\"game-modal\\">
  <div class=\\"game-modal__overlay\\"></div>
  <div class=\\"game-modal__body\\">
    <div class=\\"flex w-full h-full justify-center items-center\\">
      <div class=\\"w-full\\">
        <div class=\\"mb-4 text-secondary text-lg\\"> Hi, Welcome to Game Memory. Please type your name </div>
        <div><input class=\\"game-text-box\\" modelvalue=\\"\\" placeholder=\\"Enter your name\\"></div>
        <div class=\\"flex justify-end mt-4\\"><button role=\\"button\\" class=\\"game-button md:w-auto w-full md:w-auto w-full\\" disabled=\\"\\"> Start </button></div>
      </div>
    </div>
  </div>
</div>"
`;
